corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js

```diff
@@ -503,6 +503,16 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                     return self.case_transaction.case_type();
                 });
                 self.updatedDescription = ko.observable();
+                self.isDeprecated = ko.computed(function () {
+                    const config = self.case_transaction.caseConfig;
+                    if (self.key() !== 'name' && _(config.deprecatedPropertiesDict).has(self.caseType())) {
```

Comment:
I presume `self.key()` is the property name?

Reply:
That is correct.

Reply:
is the `has` check required? Does not it return undefined if the key is not present?

Reply:
That is a good point, this isn't strictly necessary. It would return `undefined` which would ultimately cause the next conditional to be falsey, causing the function to correctly return false.

I don't see any harm in removing this check, so will be taking it out.

Reply:
[d57981d](https://github.com/dimagi/commcare-hq/pull/34425/commits/d57981d5daf49bb502441e80c625851f04f80abb)

Reply:
Cool. I see we did it for the first check. We can't do it for the following one as well?

Reply:
Removing the second one opens the scenario where we might try and access a property from `depProps` when it is `undefined`, causing a `TypeError` exception to be thrown. 

Still, I agree we can simplify this a little. I'll remove the `has` check and replace it with `if (depProps && self.key() !== 'name')`